basics/editor-setup.md

```diff
@@ -23,37 +23,47 @@ The choice of an editor is a very personal thing. There are many
-for writing Python code. _**TODO:** add a link to the old editor setup
-tutorial here._
-
-Most of these editors lack some important features, they have so many
-features that confuse people or they aren't free. You can use these
-editors if you like them, but **these editors are BAD for getting
-started with programming**:
-
-- PyCharm
-- IDLE
-- Emacs
-- Gedit
-- Nano
-- NetBeans
-- Notepad
-- Pluma
-- Spyder
-- Vim
-- Wingware
-
-This list doesn't contain all bad editors, but these are editors that
-people often try to use. If you know a bad editor and you think I should
+The editors can be broadly divided into three:-
+
+#### The Basic Text Editors
+These editors are usually found in-built in the OS as part of its productive suite. They do not have features like
+code compilation/running, auto-completion, syntax highlighting, etc. that makes one's life easier. They are  usually used for relatively simple
+text editing and is not preferred  for coding.  
+
+A few popular ones in this category are:
+- Notepad (Windows)
+- Gedit (Linux/Ubuntu)
+- Notepad ++ (Windows)
+- Nano (Linux,Mac OS)
+
+#### Advanced Text Editors
+The text editors in this category are  mainly preferred for coding at a beginner stage. They have features like auto-completion, syntax highlighting,
+code compilation, etc. They are "smart" editors have "smart" features like Error Checking, Debugging, etc. They are relatively easy to learn and have the necessary features
+to start your programming journey.
+
+A few popular ones in this category are:
+- Visual Studio Code / VS Code (Windows/Linux/Mac OS)
+- IDLE (Usually is preinstalled with newer versions of Python) (Windows/Linux/Mac OS)
+- Thonny (Windows/Linux/Mac OS)
+- Porcupine(Windows/Linux/Mac OS)
+- Geany(Windows/Linux/Mac OS)
+
+#### Professional Level Editors / IDEs
+This category of text editors are usually professional grade pieces of software. They are mostly proprietary and paid. They have a steep
+learning curve due to host of features they have to offer. Usually `IDE`s or `Integrated Development Environment`s fall under this.
+IDEs usually consolidates the basic tools that are necessary to not only build but also test the software. These types of editors are generally not preferred
+in the beginning stage and is usually preferred when writing complex pieces of software.
+
+A few popular ones in this category are:
+- Visual Studio (Not be confused with `Visual Studio Code`) (Windows)
+- Pycharm (Windows/Linux/Mac OS)
+- Vim (Windows/Linux/Mac OS)
+- Emacs (Windows/Linux/Mac OS)
+
+There are no "right" or "wrong" editors.The preference of an editor
+is a personal choice and thus exploring all the options out there would be necessary to choose your preferred editor.
+This list doesn't contain all the editors out there but a few popular ones.
+If you know another editor that has not been included, and you think I should
 mention it here, please [let me know](../contact-me.md).
```

Comment:
I don't think we need this anymore. The lists are pretty good right now, and the bottom of the page basically says the same thing.
```suggestion
```